lib/Target/CBackend/CBackend.cpp

```diff
@@ -870,13 +871,15 @@ bool CWriter::printConstantString(Constant *C, enum OperandContext Context) {
 
 // TODO copied from CppBackend, new code should use raw_ostream
 static inline std::string ftostr(const APFloat &V) {
```

Comment:
This function seems odd, when LLVM obviously already inherently knows how to do this (to print IR):
```suggestion
static inline std::string ftostr(const APFloat &V) {
  SmallVector<char, 32> Buffer;
  V.toString(Buffer);
  return std::string(Buffer.data(), Buffer.size());
}
```

Reply:
Cool, I'll test that and update the PR.

Reply:
Works nicely.